### PR TITLE
add pyramid_retry

### PR DESCRIPTION
**Pull request by stevepiercy**
- also sort requirements in setup.py
- see https://github.com/Pylons/pyramid/issues/3042

### CODE REVIEW

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/models/__init__.py [hunk 1]
```diff
@@ -62,6 +62,9 @@ def includeme(config):
     # use pyramid_tm to hook the transaction lifecycle to the request
     config.include('pyramid_tm')
 
+    # use pyramid_retry to allow retrying N times before indicating failure
```
Comment (mmerickel) [inline {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/models/__init__.py]: pyramid_retry doesn't retry every failure... only certain types. I'm not quite comfortable with the wording here.

"use pyramid_retry to allow retrying a request after certain exceptions before failing" ?

File: {{cookiecutter.repo_name}}/setup.py [hunk 2]
```diff
@@ -9,10 +9,11 @@
     CHANGES = f.read()
 
 requires = [
-    'plaster_pastedeploy',
-    'pyramid',
-    'pyramid_jinja2',
+    'pyramid >= 1.9a',
     'pyramid_debugtoolbar',
+    'pyramid_jinja2',
+    'plaster_pastedeploy',
```
Comment (mmerickel) [inline {{cookiecutter.repo_name}}/setup.py]: alphabetical order was lost here?

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/models/__init__.py (hunk 1 above)
Comment (stevepiercy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/models/__init__.py]: Yeah, it is kinda klunky. How about this?

"use pyramid_retry to retry a request after certain exceptions occur as many as `retry.attempts` before failing"

File: {{cookiecutter.repo_name}}/setup.py (hunk 2 above)
Comment (stevepiercy) [reply {{cookiecutter.repo_name}}/setup.py]: Fixed in 64101a8

File: {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/models/__init__.py (hunk 1 above)
Comment (mmerickel) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/models/__init__.py]: "use pyramid_retry to retry a request when transient exceptions occur" should be sufficient

Comment (stevepiercy) [reply {{cookiecutter.repo_name}}/{{cookiecutter.repo_name}}/models/__init__.py]: done in 7751565

**Pull request merged**